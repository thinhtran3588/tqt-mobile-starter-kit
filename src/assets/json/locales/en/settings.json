{
  "author": "Author",
  "betaUser": "Beta user",
  "copyright": "Copyright",
  "darkTheme": "Dark theme",
  "generalInfo": "General info",
  "joinBetaUserConfirmation": "Do you want to be a beta user & get update more frequently?",
  "language": "Language",
  "leaveBetaUserConfirmation": "Do you want to leave beta program?",
  "primaryColor": "Primary color",
  "privacyPolicy": "Privacy Policy",
  "signIn": "Sign in",
  "signOut": "Sign out",
  "tos": "Terms Of Service",
  "useSystemTheme": "Use system theme",
  "version": "Version"
}
