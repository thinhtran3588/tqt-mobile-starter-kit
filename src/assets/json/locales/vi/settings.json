{
  "author": "Tác giả",
  "copyright": "Bản quyền",
  "darkTheme": "Theme tối",
  "generalInfo": "Thông tin chung",
  "primaryColor": "Màu chính",
  "useSystemTheme": "Dùng theme hệ thống",
  "version": "Phiên bản"
}
