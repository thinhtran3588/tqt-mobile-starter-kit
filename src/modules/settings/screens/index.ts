export * from './settings/settings.screen';
export * from '../../../core/screens/web-view/web-view.screen';
