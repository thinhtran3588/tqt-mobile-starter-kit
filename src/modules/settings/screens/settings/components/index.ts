export * from './theme-setting/theme-setting.component';
export * from './primary-color-setting/primary-color-setting.component';
export * from './general-info/general-info.component';
export * from './language-setting/language-setting.component';
