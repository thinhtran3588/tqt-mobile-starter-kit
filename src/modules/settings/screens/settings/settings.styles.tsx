import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  header: {
    // backgroundColor: '#fff',
  },
  item: {
    padding: 0,
  },
  itemTitle: {
    fontWeight: 'bold',
  },
  switchContainer: {
    display: 'flex',
    justifyContent: 'center',
    paddingRight: 10,
  },
});
