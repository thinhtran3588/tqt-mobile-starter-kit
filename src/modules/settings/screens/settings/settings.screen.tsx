import React from 'react';
import {useTranslation} from 'react-i18next';
import {Text, List, Layout, ScrollView} from '@core/components';
import {ThemeSetting, GeneralInfo, LanguageSetting} from './components';

export const SettingsScreen = (): JSX.Element => {
  const {t} = useTranslation('settings');
  return (
    <Layout headerTitle={t('common:settings')}>
      <ScrollView>
        <ThemeSetting />
        <LanguageSetting />
        <List.Subheader>
          <Text type='h4'>{t('generalInfo')}</Text>
        </List.Subheader>
        <GeneralInfo />
      </ScrollView>
    </Layout>
  );
};
