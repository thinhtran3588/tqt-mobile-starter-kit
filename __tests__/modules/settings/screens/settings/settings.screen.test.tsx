import 'react-native';
import React from 'react';
import {render, toJSON, fireEvent} from '@test-utils';
import {SettingsScreen} from '@settings/screens/settings/settings.screen';

it('renders correctly', async () => {
  const {baseElement} = render(<SettingsScreen />);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('uses light theme', async () => {
  const {baseElement, getByTestId} = render(<SettingsScreen />);
  fireEvent.valueChange(getByTestId('system-theme-switch'), false);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('uses dark theme', async () => {
  const {baseElement, getByTestId} = render(<SettingsScreen />);
  fireEvent.valueChange(getByTestId('system-theme-switch'), false);
  fireEvent.valueChange(getByTestId('dark-theme-switch'), true);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('opens primary color picker', async () => {
  const {baseElement, getByTestId} = render(<SettingsScreen />);
  fireEvent.press(getByTestId('primary-color-list-item'));
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('opens language picker', async () => {
  const {baseElement, getByTestId} = render(<SettingsScreen />);
  fireEvent.press(getByTestId('language-list-item'));
  expect(toJSON(baseElement)).toMatchSnapshot();
});
