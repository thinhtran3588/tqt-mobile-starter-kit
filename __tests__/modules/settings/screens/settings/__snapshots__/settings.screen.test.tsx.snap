// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "backgroundColor": "#00bcd4",
            "elevation": 4,
            "zIndex": 0,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00bcd4",
            "elevation": 0,
            "flexDirection": "row",
            "height": 56,
            "marginTop": 0,
            "paddingHorizontal": 4,
          }
        }
      >
        <View
          style={
            Object {
              "width": 48,
            }
          }
        />
        <TouchableWithoutFeedback
          disabled={true}
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 12,
                },
                Array [
                  false,
                  Object {
                    "alignItems": "center",
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              accessibilityRole="header"
              accessibilityTraits="header"
              accessible={true}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                    },
                    undefined,
                  ],
                ]
              }
            >
              common:settings
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <View
          style={
            Object {
              "width": 48,
            }
          }
        />
      </View>
    </SafeAreaView>
    <Text
      style={
        Array [
          undefined,
          Object {
            "fontSize": 12,
          },
        ]
      }
    >
      Version: 0.1.1
    </Text>
  </View>
</View>
`;
